### DISCUSSION TITLE
How to run hlo-opt?

### DISCUSSION THREAD

**Discussion by AlexRibard** [openxla/xla#18398] [General]
I've followed the instruction to build from source with docker.
And I am trying to run `hlo-opt` on some HLO file.
https://openxla.org/xla/tools
This tells me to run hlo-opt --platform=CPU --list-stages
But I just get `exec: "hlo-opt": executable file not found in $PATH: unknown`

**Comment by mooskagh** [comment]
You need to compile (or directly run) it using bazel, something like:

To run directly through bazel:
```
$ bazel run -c opt //xla/tools:hlo-opt -- --platform=CPU --list-stages
```

If you already compiled it, `hlo-opt` will be in the `./bazel-bin/xla/tools/` subdirectory of your xla repository.

**Comment by AlexRibard** [comment]
Nice! thank you it works :)
I am now trying to run hlo on a file and see the resulting optimization. For example:
`bazel run -c opt //xla/tools:hlo-opt <paht>xla/xla/service/gpu/fusions/tests/input_slices/simple.hlo -- --platform=CPU --stage=hlo`
How do I dump out the resulted optimized hlo file?

**Comment by AlexRibard** [comment]
Erratum, it worked! amazing
